sky/packages/sky/lib/src/fn3/drag_target.dart

```diff
@@ -3,15 +3,85 @@
 // found in the LICENSE file.
 
 import 'dart:collection';
+import 'dart:sky' as sky;
 
 import 'package:sky/rendering.dart';
 import 'package:sky/src/fn3/basic.dart';
 import 'package:sky/src/fn3/binding.dart';
 import 'package:sky/src/fn3/framework.dart';
+import 'package:sky/src/fn3/navigator.dart';
 
 typedef bool DragTargetWillAccept<T>(T data);
 typedef void DragTargetAccept<T>(T data);
 typedef Widget DragTargetBuilder<T>(BuildContext context, List<T> candidateData, List<dynamic> rejectedData);
+typedef void DragFinishedNotification();
+
+class Draggable extends StatefulComponent {
+  Draggable({ Key key, this.navigator, this.data, this.child, this.feedback }): super(key: key) {
+    assert(navigator != null);
+  }
+  final NavigatorState navigator;
+  final dynamic data;
+  final Widget child;
+  final Widget feedback;
+  DraggableState createState() => new DraggableState();
```

Comment:
In other places, we've been using a blank line after the constructor and before the createState.  Not sure if that makes much of a difference.